tests/dialects/test_riscv.py

```diff
@@ -35,7 +34,7 @@ def test_add_op():
     assert a2.type.index == IntAttr(12)
 
     # Registers that aren't predefined should not have an index.
-    assert isinstance(riscv.IntRegisterType.infinite_register(1).index, NoneAttr)
+    assert riscv.IntRegisterType.infinite_register(1).index == IntAttr(-2)
```

Comment:
Why do I have to keep track that 1 is -2, instead of a straightforward match from 1 to -1, 2 to -2, etc.?
The 0 index is taken, and this infinite register set doesn't overlap with anything else?

What is the point of being subjected to this mild number juggling? What does it save us?

Reply:
Hmm, do you have to keep track of it? To me it seems like an implementation detail of the storage of negative indices, it doesn't really matter much whether it's -1 or -2, right? It's never shown in the IR, and users are never expected to construct the attribute using the init, as opposed to one of the `from_` helpers.

Reply:
Even more so then, Why do this? If I visited this codebase for the first time, I'd stumble in comprehension here

Reply:
OK so if I understand correctly, you would propose for the infinite registers to not start at `j0`, but at `j1`, where the index attribute would hold `-1`?

Reply:
Actually, did we change this recently, or I misremember (the start of the j registers I mean)? If so, then maybe that is too much flip-flopping

Reply:
The thing I changed recently was that int and float riscv registers actually check that they're the correct (j/fj) prefix, not really the numbering.

Reply:
I'm happy to change the 0- or 1-indexed infinite registers, I don't feel strongly either way, we just need some index-based representation.

Reply:
hmm, I changed my mind, let's just keep it that way, starting from 0 for infinite registers raises less questions and it seems more natural, following actual ISAs.